dwave/system/composites/reversecomposite.py

```diff
@@ -0,0 +1,204 @@
+# Copyright 2019 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# =============================================================================
+"""
+Composite that do batch operations on initial samples provided for reverse annealing.
+"""
+import dimod
+import numpy as np
+
+__all__ = 'ReverseAdvanceComposite', 'BatchReverseComposite'
+
+
+class ReverseAdvanceComposite(dimod.ComposedSampler):
+    """ Composite that advances a sample using reverse annealing along a
+        given set of anneal schedules.
+
+    Args:
+       sampler (:obj:`dimod.Sampler`):
+            A dimod sampler
+
+    """
+
+    def __init__(self, child_sampler):
+        self._children = [child_sampler]
+
+    @property
+    def children(self):
+        return self._children
+
+    @property
+    def parameters(self):
+        param = self.child.parameters.copy()
+        param['schedules'] = []
+        return param
+
+    @property
+    def properties(self):
+        return {'child_properties': self.child.properties.copy()}
+
+    def sample(self, bqm, anneal_schedules=None, **parameters):
+        """ Composite that advances a sample using reverse annealing along a
+        given set of anneal schedules. Always selects the most probable lowest energy sample out of
+        the previous steps return
+
+        Args:
+            bqm (:obj:`dimod.BinaryQuadraticModel`):
+                Binary quadratic model to be sampled from.
+
+            schedules (list): an ordered list of anneal schedules. The first element of the
+                list should correspond to the first anneal schedule.
+
+            **parameters:
+                Parameters for the sampling method, specified by the child sampler.
+
+        Returns:
+            :obj:`dimod.SampleSet` that has initial_state and schedule_index fields in addition to
+            'energy' and 'num_occurrences'
```

Comment:
The fields specified after "in addition to" are either incomplete (missing "sample") or it's unclear why you are specifying these, I think you want to call attention to the answer_mode as histogram and for that you say just num_occurrences but maybe better to talk about that separately?